### PR TITLE
[REGEDIT] Fix issue at Find registry key

### PR DESCRIPTION
**Pull request by zecarlos1957**
Searching the registry for a key or value gives no results 

[REGEDIT] Fix issue at Find registry key
If we dont select a value to [pszValueName] as a starting point to search from,  we need set [pszValueName]  to the first item in current subkey

[[REGEDIT] Set focus to subkey when we search for it.

[[REGEDIT] Check pszSubKey length before calling RegFindRecurse.

### CODE REVIEW

**Comment by zecarlos1957** [comment]
If we are looking for a non-existent string and press F3 key, this give a error: 
[The instruction at [addr] referenced memory at  [addr]. The memory could not be read]. 
This issue is not in RegEdit code. I tested our RegEdit.exe in WinXP and works fine with no errors.

File: base/applications/regedit/find.c [hunk 1]
```diff
@@ -370,13 +373,13 @@ BOOL RegFindWalk(
     LPWSTR *ppszNames = NULL;
 
     hBaseKey = *phKey;
+    if (wcslen(pszSubKey) >= MAX_PATH)
```
Comment (HBelusca) [inline base/applications/regedit/find.c:376]:
```suggestion

    if (wcslen(pszSubKey) >= _countof(szSubKey))
```

File: base/applications/regedit/find.c [hunk 2]
```diff
@@ -687,10 +690,17 @@ BOOL FindNext(HWND hWnd)
...
-        SetValueName(g_pChildWnd->hListWnd, pszFoundValueName);
+
+        if (pszFoundValueName != NULL)
+        {
+            SetValueName(g_pChildWnd->hListWnd, pszFoundValueName);
+            free(pszFoundValueName);
+            SetFocus(g_pChildWnd->hListWnd);
+        }
+        else
+            SetFocus(g_pChildWnd->hTreeWnd);
```
Comment (HBelusca) [inline base/applications/regedit/find.c:701]:
```suggestion
        else
        {
            SetFocus(g_pChildWnd->hTreeWnd);
        }
```

File: base/applications/regedit/find.c [hunk 3]
```diff
@@ -687,10 +690,17 @@ BOOL FindNext(HWND hWnd)
     {
         GetKeyName(szFullKey, COUNT_OF(szFullKey), hKeyRoot, pszFoundSubKey);
         SelectNode(g_pChildWnd->hTreeWnd, szFullKey);
-        SetValueName(g_pChildWnd->hListWnd, pszFoundValueName);
+
```
Comment (HBelusca) [inline base/applications/regedit/find.c:693]:
```suggestion
        free(pszFoundSubKey);

```

File: base/applications/regedit/find.c [hunk 4]
```diff
@@ -687,10 +690,17 @@ BOOL FindNext(HWND hWnd)
...
+        if (pszFoundValueName != NULL)
+        {
+            SetValueName(g_pChildWnd->hListWnd, pszFoundValueName);
+            free(pszFoundValueName);
+            SetFocus(g_pChildWnd->hListWnd);
+        }
+        else
+            SetFocus(g_pChildWnd->hTreeWnd);
+
         free(pszFoundSubKey);
```
Comment (HBelusca) [inline base/applications/regedit/find.c:703]:
```suggestion
```

File: base/applications/regedit/find.c [hunk 5]
```diff
@@ -195,6 +195,9 @@ BOOL RegFindRecurse(
 
     qsort(ppszNames, c, sizeof(LPWSTR), compare);
 
+    if (pszValueName == s_empty)
```
Comment (HBelusca) [inline base/applications/regedit/find.c:198]: It looks to me that you could even delete the previous `if (pszValueName == NULL) ...` case (that makes the pointer point to `s_empty`, and directly change here your new 'if'-case by `if (pszValueName == NULL)  pszValueName = ppszNames[0];`.

**Review by HBelusca — APPROVED** [review]

**Pull request merged**